src/Interface/chiopInterface.cpp

```diff
@@ -0,0 +1,36 @@
+#include "chiopInterface.hpp"
+extern "C" {
+
+using namespace hiop;
+
+// These are default options for the C interface for now. Setting options from C will be added in the future.
+int hiop_createProblem(cHiopProblem *prob) {
+  cppUserProblem * cppproblem = new cppUserProblem(prob);
```

Comment:
please also delete this; it leaks.

I did by adding a `void *` member in the `C` struct to hold this, and then `static_cast` and `delete` in `hiop_destroyProblem` (I couldn't push because I don't have rights to your fork, maybe next time you should branch of this repo?)